sdk/device_update/src/client.rs

```diff
@@ -94,23 +98,28 @@ impl DeviceUpdateClient {
             req = req.header("Content-Length", 0);
         }
 
-        let resp = req
-            .send()
-            .await
-            .map_err(|e| Error::Core(CoreError::Http(HttpError::ExecuteRequest(e.into()))))?;
+        let resp = req.send().await.with_context(ErrorKind::Io, || {
+            format!("failed to send request. uri: {uri}")
+        })?;
 
         if resp.status() == 202u16 {
             let headers = resp.headers();
             return match headers.get("operation-location") {
-                Some(location) => location
-                    .to_str()
-                    .map(|x| x.to_string())
-                    .map_err(|_| Error::InvalidOperationPath),
-                None => Err(Error::NoOperationLocation),
+                Some(location) => location.to_str().map(|x| x.to_string()).map_err(|_| {
+                    Error::with_message(ErrorKind::Other, || {
+                        "successful import (202 status) but no operation-location header found"
+                            .to_string()
```

Comment:
This error needs to be swapped with the following one - i.e. this one should be "invalid characters..."

Reply:
Sure